flytekit/clis/sdk_in_container/serialize.py

```diff
@@ -180,30 +182,30 @@ def serialize_all(
             if isinstance(entity, PythonTask) or isinstance(entity, Workflow) or isinstance(entity, LaunchPlan):
                 if isinstance(entity, PythonTask):
                     if mode == SerializationMode.DEFAULT:
-                        serializable = get_serializable(ctx.serialization_settings, entity)
+                        get_serializable(new_api_serializable_entities, ctx.serialization_settings, entity)
                     elif mode == SerializationMode.FAST:
-                        serializable = get_serializable(ctx.serialization_settings, entity, fast=True)
+                        get_serializable(new_api_serializable_entities, ctx.serialization_settings, entity, fast=True)
                     else:
                         raise AssertionError(f"Unrecognized serialization mode: {mode}")
-                    loaded_entities_types[ResourceType.TASK].append(serializable)
                 else:
-                    serializable = get_serializable(ctx.serialization_settings, entity)
+                    get_serializable(new_api_serializable_entities, ctx.serialization_settings, entity)
+                # loaded_entities.append(serializable)
+
                 if isinstance(entity, Workflow):
-                    loaded_entities_types[ResourceType.WORKFLOW].append(serializable)
                     lp = LaunchPlan.get_default_launch_plan(ctx, entity)
-                    launch_plan = get_serializable(ctx.serialization_settings, lp)
-                    loaded_entities_types[ResourceType.LAUNCH_PLAN].append(launch_plan)
-                if isinstance(entity, LaunchPlan):
-                    loaded_entities_types[ResourceType.LAUNCH_PLAN].append(serializable)
+                    get_serializable(new_api_serializable_entities, ctx.serialization_settings, lp)
 
+        all_entities = list(new_api_serializable_entities.values())
+        registerable_entities = list(filter(lambda x: isinstance(x, RegisterableEntity), all_entities))
```

Comment:
is this filter necessary? i thought all FlyteEntities were registerable

Reply:
nope, will explain in comment.  sdknode or example shows up